src/libs/ReportActionsUtils.js

```diff
@@ -547,6 +547,17 @@ function getIOUReportIDFromReportActionPreview(reportAction) {
     return lodashGet(reportAction, 'originalMessage.linkedReportID', '');
 }
 
+/**
+ * Find the reportAction whose childReportID is taskReportID in reportActions with reportID
+ *
+ * @param {String} taskReportID
+ * @param {String} reportID
+ * @returns {Object}
+ */
+function getParentReportActionForTask(taskReportID, reportID) {
+    return _.find(allReportActions[reportID], (reportAction) => reportAction && `${reportAction.childReportID}` === `${taskReportID}`);
```

Comment:
> @s-alves10 Let's write the code as if they are both strings. Parent report ids being returned as a number is a bug https://expensify.slack.com/archives/C01GTK53T8Q/p1691167905181059?thread_ts=1691167905.181059&cid=C01GTK53T8Q

@s77rt 

Got it. But if we don't convert them to string, comparison would return false and can't find the parent action in assignee report. `childReportID` is a number as well.

I think we need to convert them to string unless the backend is fixed. What do you think?

Reply:
Indeed. We may need to wait for an outcome from that slack thread. But for now let's write the right code.